### PR TITLE
[SUPPORT]: Remove code smells + add code coverage + update Jest Setup ✅

### PR DESCRIPTION
**Pull request by mcayuelas-ledger**
### 📝 Description

This PR aims to improve code quality and increase the test coverage of the current code. 
 - Some changes in the Jest setups have been made to avoid the need to mock various libraries every time.
### ❓ Context

- **Linked resource(s)**: `` 

### 🚀 Expectations to reach

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by cgrellard-ledger — APPROVED** [review]

File: src/data/__tests__/networkConfig.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    expect(base).toBeDefined();
+    expect(base?.displayName).toBe("Base");
+  });
+});
+
+describe("EIP155_GOERLI_CHAINS", () => {
+  it("contains Ethereum Goerli", () => {
+    const ethereumGoerli: Network | undefined = SUPPORTED_NETWORK["ethereum_goerli"];
+    expect(ethereumGoerli).toBeDefined();
+    expect(ethereumGoerli?.displayName).toBe("Ethereum Goerli");
```
Comment (KVNLS) [inline src/data/__tests__/networkConfig.test.ts:52]: You can combine both assertions with only `expect(ethereumGoerli?.displayName).toBe("Ethereum Goerli");`
The less we write code, the less we have to maintain or change it.
The second assertion is what you want to test and includes the first one.

File: src/__tests__/units/components/walletConnectPopin.test.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  it("should call the onClose function when the close button is clicked", () => {
+    render(
+      <WalletConnectPopin isOpen={true} onClose={mockOnClose}>
+        <div>Content goes here</div>
+      </WalletConnectPopin>,
+    );
+
+    const closeButton = screen.getByTestId("close-button");
+    fireEvent.click(closeButton);
```
Comment (KVNLS) [inline src/__tests__/units/components/walletConnectPopin.test.tsx:39]: I think that we should `userEvent` here, is there any issue with it?

File: src/__tests__/units/tabs.test.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    // Check if the disabled tab has the "disabled" class
+    expect(screen.getByText("Tab 3").closest("div")).toHaveStyle("cursor: not-allowed");
+  });
+
+  it("calls setActiveTabIndex when a tab is clicked", () => {
+    renderTabs();
+
+    // Click on the second tab (Tab 2)
+    fireEvent.click(screen.getByText("Tab 2"));
```
Comment (KVNLS) [inline src/__tests__/units/tabs.test.tsx:55]: `userEvent`

**Review by KVNLS — CHANGES_REQUESTED** [review]
Great work!

File: src/data/__tests__/networkConfig.test.ts (hunk 1 above)
Comment (mcayuelas-ledger) [reply src/data/__tests__/networkConfig.test.ts:52]: Yes indeed !

**Review by KVNLS — APPROVED** [review]
🎉

**Pull request merged**